exporter/azuremonitorexporter/traceexporter.go

```diff
@@ -0,0 +1,535 @@
+			data.Properties[k] = stringValue
+		}
+	}
+
+	if httpURL != "" {
+		if u, err := url.Parse(httpURL); err == nil {
+			data.Name = httpMethod + " " + u.Path
+			data.Data = httpURL
+			data.Target = u.Host
+		}
+	} else if httpScheme != "" && httpHost != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + httpHost + httpTarget
+		data.Target = httpHost
+	} else if httpScheme != "" && peerHostName != "" && peerPort != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + peerHostName + ":" + peerPort + httpTarget
+		data.Target = peerHostName + ":" + peerPort
+	} else if httpScheme != "" && peerIP != "" && peerPort != "" && httpTarget != "" {
+		data.Name = httpMethod + " " + httpTarget
+		data.Data = httpScheme + "://" + peerIP + ":" + peerPort + httpTarget
+		data.Target = peerIP + ":" + peerPort
+	}
+}
+
+// Sets properties on an AppInsights RemoteDependencyData from the wire format Span when the outbound Span type is gRPC
+func fillRemoteDependencyDataGrpc(span *tracepb.Span, data *contracts.RemoteDependencyData) {
+	data.Type = "Grpc"
+
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
+	peerService := ""
+	peerHostName := ""
+	peerPort := ""
+
+	for k, v := range span.Attributes.AttributeMap {
+		stringValue := attributeValueAsString(v)
+		switch k {
+		case spanAttributeKeyComponent:
+			// do nothing
+		case spanAttributeKeyRPCStatusCode:
+			data.ResultCode = stringValue
+			code, err := strconv.Atoi(data.ResultCode)
+			if err == nil {
+				data.Success = code == int(codes.OK)
+			}
+		case spanAttributeKeyPeerService:
+			peerService = stringValue
+		case spanAttributeKeyPeerHostname:
+			peerHostName = stringValue
+		case spanAttributeKeyPeerPort:
+			peerPort = stringValue
+		default:
+			// Anything not on this list above ends up as a custom property
+			data.Properties[k] = stringValue
+		}
+	}
+
+	if peerService != "" && peerHostName != "" && peerPort != "" {
+		data.Data = peerHostName + peerPort + peerService
+		data.Target = peerHostName + peerPort
+	}
+}
+
+func prefixIfNecessary(s string, prefix string) string {
+	if strings.HasPrefix(s, prefix) {
+		return s
+	}
+
+	return prefix + s
+}
+
+func (exporter *traceExporter) pushTraceData(
+	context context.Context,
+	traceData consumerdata.TraceData,
+) (droppedSpans int, err error) {
+
+	spanCount := len(traceData.Spans)
+	if spanCount == 0 {
+		return 0, nil
+	}
+
+	for _, wireFormatSpan := range traceData.Spans {
+		if envelope, err := spanToEnvelope(exporter.config.InstrumentationKey, wireFormatSpan); err == nil && exporter.transportChannel != nil {
+			// This is a fire and forget operation
+			exporter.transportChannel.Send(envelope)
+		} else {
+			// Only tracks the inability to transform a wire format Span to an AppInsights envelope
+			droppedSpans++
```

Comment:
should add metrics to drops and other stats

Reply:
> should add metrics to drops and other stats

I think any metrics for dropped spans would/should be emitted by the code that calls this method. That code exists in the main collector repo.

The package that defines the Send method treats it as fire and forget so I wouldn't know if the Spans are delivered or not at this point; I wold have to capture and parse the debug output or add more structured instrumentation that package.